sql/pgwire: TestParseClientProvidedSessionParameters failed

sql/pgwire.TestParseClientProvidedSessionParameters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7941380?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7941380?buildTab=artifacts#/) on master @ [99e7bd7e72741106fb6da9cc1fbbaf35bc7b9fde](https://github.com/cockroachdb/cockroach/commits/99e7bd7e72741106fb6da9cc1fbbaf35bc7b9fde):


```
        	Error:      	Expect "unexpected protocol version: 80877102" to match "remote port is not numeric"
        	Test:       	TestParseClientProvidedSessionParameters/remote_addr_port_must_be_numeric
    --- FAIL: TestParseClientProvidedSessionParameters/remote_addr_port_must_be_numeric (0.00s)
=== RUN   TestParseClientProvidedSessionParameters/success_setting_remote_address_from_query
    conn_test.go:1539: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2281/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/pgwire/pgwire_test_/pgwire_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/pgwire/conn_test.go:1539
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2281/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/pgwire/pgwire_test_/pgwire_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/pgwire/conn_test.go:1584
        	Error:      	Received unexpected error:
        	            	unexpected protocol version: 80877102
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/pgwire.getSessionArgs
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn_test.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/pgwire.TestParseClientProvidedSessionParameters.func26
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn_test.go:1583
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) unexpected protocol version: 80877102
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestParseClientProvidedSessionParameters/success_setting_remote_address_from_query
    --- FAIL: TestParseClientProvidedSessionParameters/success_setting_remote_address_from_query (0.00s)
=== RUN   TestParseClientProvidedSessionParameters/crdb:jwt_auth_enabled_must_be_a_boolean_if_provided
    conn_test.go:1508: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2281/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/pgwire/pgwire_test_/pgwire_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/pgwire/conn_test.go:1508
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2281/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/pgwire/pgwire_test_/pgwire_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/pgwire/conn_test.go:1584
        	Error:      	Expect "unexpected protocol version: 80877102" to match "crdb:jwt_auth_enabled: strconv.ParseBool: parsing "notaboolean": invalid syntax"
        	Test:       	TestParseClientProvidedSessionParameters/crdb:jwt_auth_enabled_must_be_a_boolean_if_provided
    --- FAIL: TestParseClientProvidedSessionParameters/crdb:jwt_auth_enabled_must_be_a_boolean_if_provided (0.00s)
=== RUN   TestParseClientProvidedSessionParameters/normalize_to_lower_case_in_options_parameter
    conn_test.go:1547: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2281/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/pgwire/pgwire_test_/pgwire_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/pgwire/conn_test.go:1547
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2281/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/pgwire/pgwire_test_/pgwire_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/pgwire/conn_test.go:1584
        	Error:      	Received unexpected error:
        	            	unexpected protocol version: 80877102
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/pgwire.getSessionArgs
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn_test.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/pgwire.TestParseClientProvidedSessionParameters.func26
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn_test.go:1583
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) unexpected protocol version: 80877102
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestParseClientProvidedSessionParameters/normalize_to_lower_case_in_options_parameter
    --- FAIL: TestParseClientProvidedSessionParameters/normalize_to_lower_case_in_options_parameter (0.00s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-experience @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParseClientProvidedSessionParameters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/pgwire/conn_test.go
+++ b/pkg/sql/pgwire/conn_test.go
@@ -42,6 +42,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/tests"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -1336,6 +1337,7 @@ func TestConnServerAbortsOnRepeatedErrors(t *testing.T) {
 
 func TestParseClientProvidedSessionParameters(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	skip.WithIssue(t, 93469, "flaky test")
 	defer log.Scope(t).Close(t)
 
 	// Start a pgwire "server".

